tests/sdk/mcp/test_mcp_tool_validation.py

```diff
@@ -0,0 +1,54 @@
+import mcp.types
+import pytest
+from unittest.mock import Mock
+
+from openhands.sdk.mcp.tool import MCPTool
+from openhands.sdk.mcp.client import MCPClient
+
+
+def _make_tool_with_schema(schema: dict):
+    mcp_tool = mcp.types.Tool(
+        name="fetch",
+        description="Fetch a URL",
+        inputSchema=schema,
+    )
+    client = Mock(spec=MCPClient)
+    return MCPTool.create(mcp_tool, client)[0]
+
+
+def test_mcp_action_from_arguments_validates_and_sanitizes():
+    tool = _make_tool_with_schema(
+        {
+            "type": "object",
+            "properties": {
+                "url": {"type": "string"},
+                "timeout": {"type": "number"},
+            },
+            "required": ["url"],
+        }
+    )
+
+    # includes an extra field that should be rejected and a None that should be dropped
```

Comment:
```suggestion
    # includes a None that should be dropped
```

I don't think there's an extra field that should be rejected here